main.tf

```
@@ -46,3 +46,41 @@ module "s3_user" {
   s3_actions   = ["${var.allowed_bucket_actions}"]
   s3_resources = ["${join("", aws_s3_bucket.default.*.arn)}/*", "${join("", aws_s3_bucket.default.*.arn)}"]
 }
+
+resource "aws_s3_bucket_policy" "default" {
+  count  = "${var.allow_encrypted_uploads_only == "true" ? 1 : 0}"
+  bucket = "${aws_s3_bucket.default.id}"
```

```suggestion
  bucket = "${join("", aws_s3_bucket.default.*.id)}"
```

Use `splat+join` pattern when accessing resources with `count` since having `count1 makes a resource a list.
To access elements from a list, use `splat` operator (`.*.`).
To prevent TF errors when `count=0`, use `join` (which will produce an empty string when `count=0`.

---

done